Zend/zend_compile.c

```diff
@@ -6495,7 +6532,34 @@ static void zend_is_type_list_redundant_by_single_type(zend_type_list *type_list
 		}
 		if (zend_string_equals_ci(ZEND_TYPE_NAME(type_list->types[i]), ZEND_TYPE_NAME(type))) {
 			zend_string *single_type_str = zend_type_to_string(type);
-			zend_error_noreturn(E_COMPILE_ERROR, "Duplicate type %s is redundant", ZSTR_VAL(single_type_str));
+			if (
```

Comment:
Yes, because otherwise you would get an error message like ``Duplicate type T is redundant`` when ``self`` or ``parent`` gets resolved to ``T`` and it makes for some confusing stuff, or I'm misunderstanding the question.